src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -84,6 +85,13 @@
   private final AzureBlobStorePdao azureBlobStorePdao;
   private final GcsProjectFactory gcsProjectFactory;
 
+  private final Map<UUID, SnapshotProject> snapshotProjects =
+      Collections.synchronizedMap(new PassiveExpiringMap<>(15, TimeUnit.MINUTES));
+  private final Map<UUID, SnapshotCacheResult> snapshotCache =
```

Comment:
I was kinda wondering: would it make sense to have this live in the snapshot service?  Other places could perhaps benefit from this

Reply:
I thought about that, but the only time we're really seeing this type of traffic is in the Drs object lookups. And even then, we're only using 3 fields out of the massive Snapshot object. So, just grabbing the 3 fields from the Snapshot that we need, and caching that helps keep the cache size small, gives us more room to grow it, and should be able to be better-optimized by the compiler. 

I think there's definitely a benefit to caching at a lower layer, but I think its a much larger effort, and this small cache gets us 90% of the way there for 10% of the effort.

Reply:
Got it.  That makes a ton of sense